src/test/java/com/amplitude/AmplitudeMultiThreadTest.java

```diff
@@ -0,0 +1,211 @@
+package com.amplitude;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.amplitude.exception.AmplitudeInvalidAPIKeyException;
+import com.amplitude.util.EventsGenerator;
+
+@ExtendWith(MockitoExtension.class)
+public class AmplitudeMultiThreadTest {
+  private final String apiKey = "test-apiKey";
+  private AtomicInteger totalEvents;
+  private AtomicInteger successEvents;
+  private AtomicInteger failedEvents;
+  private AtomicInteger pendingEvents;
+
+  @BeforeEach
+  public void setUp() {
+    totalEvents = new AtomicInteger();
+    successEvents = new AtomicInteger();
+    failedEvents = new AtomicInteger();
+    pendingEvents = new AtomicInteger();
+  }
+
+  @Test
+  public void testMultipleThreadLogAndSendEventSuccess()
+      throws NoSuchFieldException, IllegalAccessException, AmplitudeInvalidAPIKeyException {
+    Amplitude amplitude = Amplitude.getInstance("Multiple threads log event success.");
+    amplitude.init(apiKey);
+    HttpCall httpCall = getMockHttpCall(amplitude);
+    Response response = ResponseUtil.getSuccessResponse();
+    when(httpCall.makeRequest(anyList())).thenAnswer(invocation -> response);
+    int total = 1000;
+    CountDownCallbacks callbacks = new CountDownCallbacks(new CountDownLatch(total));
+    amplitude.setCallbacks(callbacks);
+    int threads = 10;
+    CountDownLatch threadLatch = new CountDownLatch(threads);
+    ExecutorService executorService = Executors.newFixedThreadPool(threads);
+
+    for (int i = 0; i < threads; i++) {
+      executorService.execute(new LogEventRunnable(amplitude, total / threads, threadLatch));
+    }
+
+    try {
+      threadLatch.await();
+      callbacks.blockWithTimeout(15);
```

Comment:
Why do you need this timeout?

Reply:
We need this timeout so when there is any issues the tests won't just hang there..

Reply:
Oh got it. I thought it was an additional wait, not a timeout. Makes sense.